src/main/java/com/rackspace/ceres/app/web/AbstractRestExceptionHandler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Rackspace US, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.rackspace.ceres.app.web;
+
+import com.rackspace.ceres.app.model.ApiErrorResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class AbstractRestExceptionHandler {
+
+  @ExceptionHandler(IllegalArgumentException.class)
+  public ResponseEntity<?> handleIllegalArgumentException(IllegalArgumentException e) {
+    return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+        .body(mapToApiErrorResponse(HttpStatus.BAD_REQUEST
+            .value(), e.getMessage()));
+  }
+
+  private ApiErrorResponse mapToApiErrorResponse(int status, String message) {
+    return new ApiErrorResponse().setStatus(status).setMessage(message);
```

Comment:
This loses a lot of details provided by Spring Boot's `ErrorAttributes`. Can you bring over some or all of this logic:

https://github.com/racker/salus-common/blob/master/src/main/java/com/rackspace/salus/common/web/AbstractRestExceptionHandler.java#L96-L112

Reply:
The ControllerAdvice for spring webflux does not take `ServletRequest` as it spring mvc specific. I was looking for a way to get `ServerRequest` in advice method but couldn't find anything.

Reply:
I'm not surprised that class doesn't exist, but you'll need to poke around in code like `DefaultErrorWebExceptionHandler` to see how webflux does `ErrorAttributes`.

Reply:
...unfortunately, that's how I had to discover the code needed over in salus-common